### PR TITLE
ENG-2970 Only stalls longer then 500ms are log failures

### PR DESCRIPTION
**Pull request by Scarjit**
## Summary by CodeRabbit

- **New Features**
  - Enhanced log failure detection with a more extensible system for identifying Redpanda failures, including detection for "Address already in use" and reactor stalls exceeding 500 ms.

- **Tests**
  - Added comprehensive tests for new failure detection logic, covering various scenarios and edge cases.
  - Updated existing tests to match the revised log message formats.

### CODE REVIEW

File: umh-core/pkg/service/redpanda/redpanda_log_failures.go [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+
+func (a *AddressAlreadyInUseFailure) IsFailure(logLine string) bool {
+	return strings.Contains(logLine, "Address already in use")
+}
+
+type ReactorStalledFailure struct{}
+
+var reactorStallRegex = regexp.MustCompile(`Reactor stalled for (\d+) ms`)
+
+func (r *ReactorStalledFailure) IsFailure(logLine string) bool {
```
Comment (JeremyTheocharis) [inline umh-core/pkg/service/redpanda/redpanda_log_failures.go:42]: can you add a docstring here explaining the function?

File: umh-core/pkg/service/redpanda/redpanda_log_failures.go [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+}
+
+var RedpandaFailures = []RedpandaFailure{
+	&AddressAlreadyInUseFailure{},
+	&ReactorStalledFailure{},
+}
+
+type AddressAlreadyInUseFailure struct{}
+
+func (a *AddressAlreadyInUseFailure) IsFailure(logLine string) bool {
```
Comment (JeremyTheocharis) [inline umh-core/pkg/service/redpanda/redpanda_log_failures.go:34]: here a docstring as well

File: umh-core/pkg/service/redpanda/redpanda.go [hunk 3]
```diff
@@ -854,9 +849,8 @@ func (s *RedpandaService) IsLogsFine(logs []s6service.LogEntry, currentTime time
 			continue
 		}
 
-		for _, failure := range failures {
-			if strings.Contains(log.Content, failure) {
-
+		for _, failureDetector := range RedpandaFailures {
```
Comment (JeremyTheocharis) [inline umh-core/pkg/service/redpanda/redpanda.go:852]: can you explain how this is working? so that there is in redpanda_log_failures RedpandaFailures declared, with failure detectors, which all need to implement a IsFailure function

doesn't have to be long but some documentation here would be helpful

**Review by kanapuli — APPROVED** [review]

File: umh-core/pkg/service/redpanda/redpanda_log_failures.go [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+		return false
+	}
+
+	// Convert the number of milliseconds to an integer
+	ms, err := strconv.Atoi(matches[1])
+	if err != nil {
+		return false
+	}
+
+	// Return failure if the reactor stalled for more than 500ms
```
Comment (JeremyTheocharis) [inline umh-core/pkg/service/redpanda/redpanda_log_failures.go:82]: why 500ms? --> some explanations that small reactor stalls can happen, but we should only consdier them important if they exceed a certain range. 500ms is random for now.

**Review by JeremyTheocharis — CHANGES_REQUESTED** [review]

**Review by JeremyTheocharis — APPROVED** [review]
